src/charm.py

```diff
@@ -76,6 +105,70 @@ def _pebble_checks(self) -> Dict[str, Any]:
         }
         return checks
 
+    def _prometheus_scrape(self, config):
```

Comment:
I don't think `_prometheus_scrape` and `_prometheus_remote_write` should be in `charm.py`; the whole point of having a config manager is to let it handle config stuff and keep it out of the charm

As a consequence, I would move updating alert rules outside this function, and call it directly in `reconcile` instead.

Reply:
I understand `config.py` is a "generic" config builder, and most of this method is the receiver config, but if it doesn't belong to either maybe we should think about a third place to put this in? 

I think we only need to discuss this in a meeting for me to better understand the design